legate/driver/command.py

```diff
@@ -395,6 +386,43 @@ def cmd_user_opts(
     cmd_log_levels,
     cmd_log_file,
     cmd_eager_alloc,
-    # Append user flags so they can override whatever we provided
-    cmd_user_opts,
+)
+
+CMD_PARTS_LEGION = (
```

Comment:
We should document somehow which options are only available under "legion_python" mode. Possibly these options can be listed under a separate header on the `legate --help` output.

Reply:
Unless I'm mistaken, all of the options are available via `legate` and only some are not available, when using "plain" python. Are you saying `legate --help` should be responsible for pointing out options that are not available when _not_ using `legate`? I guess I don't think `legate --help` needs to be responsible for non-`legate` usage.

Reply:
That's fine, but in that case the documentation for common options will need to be duplicated somewhere. I was thinking that we could just tell the users in the README that when they use a legate library some options will be consumed, and they should run `legate --help` to see more.

Reply:
Can we make a separate issue to discuss/resolve before 23.03? I do feel like saying "here are options you can't use when are not running legate" is confusing in and of itself, so I'd like to think on it a bit more.

Reply:
I am thinking about doing something like this:
```
$ legate --help
...
  --gdb                 run Legate inside gdb (default: False), not supported with default python interpreter. 
...
```
We add a sentence to flags that are not supported by canonical python mode.

Reply:
I would be OK with that.

Reply:
Just want to clarify how should we name the canonical python mode? plain python, canonical python or default python? In the current readme, we call it "default python" https://github.com/nv-legate/legate.core/blob/branch-22.12/README.md?plain=1#L249-L250

Reply:
Would defer to @manopapad but I will register my opinion that I don't think the phrase "canonical python" will carry an immediate, obvious meaning for typical python users.

Reply:
I agree with @bryevdv, "standard python interpreter" is probably the best

Reply:
I have added `[legate-only, not supported with standard Python invocation]` to `legate --help` https://github.com/nv-legate/legate.core/pull/539/commits/92d8f3fb14b6c36c6da0c67c5561e00af25a843e